contracts/contracts/liquidity/LiquidityReward.sol

```diff
@@ -0,0 +1,370 @@
+pragma solidity 0.5.11;
+
+import {
+    Initializable
+} from "@openzeppelin/upgrades/contracts/Initializable.sol";
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import { SafeMath } from "@openzeppelin/contracts/math/SafeMath.sol";
+import { StableMath } from "../utils/StableMath.sol";
+import { Governable } from "../governance/Governable.sol";
+
+//
+// LiquidityReward contract doles out reward for liquidity
+//   base off of Sushiswap's MasterChef: https://github.com/sushiswap/sushiswap/blob/master/contracts/MasterChef.sol
+//
+contract LiquidityReward is Initializable, Governable {
+    using SafeMath for uint256;
+    using StableMath for uint256;
+    using SafeERC20 for IERC20;
+
+    // Info of each user.
+    struct UserInfo {
+        uint256 amount; // How many LP tokens the user has provided.
+        int256 rewardDebt; // Reward debt. See explanation below.
```

Comment:
Should this be int256? Do we ever expect a negative rewardDebt?

Reply:
yes, if LP are withdraw but the reward isn't claimed then you have negative debt.